lib/pbench/agent/tool_data_sink.py

```diff
@@ -1889,7 +1889,15 @@ def driver(
     if logger is None:
         logger = get_logger(PROG, level=parsed.level)
 
-    logger.debug("params_key (%s): %r", parsed.key, tdsp)
+    logger.debug(
+        "params_key (%s): %r",
+        parsed.key,
+        {
+            k: v
+            for k, v in sorted(tdsp._asdict().items())
+            if isinstance(v, (dict, list, tuple, str, int, float, True, False, None))
+        },
```

Comment:
Why not just put an `__str__` on the `NamedTuple` and do this once rather than dragging it out here twice?

Reply:
That's a very, very good question.  😁

Reply:
OK, I did it.  I'm not completely happy with the result, but it's probably much better than doing it in-line.  I ran into a challenge over the dichotomy between `__str__()` and `__repr__()`, but, owing to the difference in their respective semantics, I decided to stick with `__str__()` and change the call site to invoke it instead of the other one.

Oh, and, FWIW, we don't get to "do this once" (unless we refactor my change and put it into a utility function (or a superclass), which we may well want to do...), because the two references go to two different classes, so I need two separate `__str__()` implementations....

Reply:
OK, we now have two stub `__str__()` functions which call a utility function to do the real work, and it seems adequate for the moment.